straxen/plugins/peaks/peaks_som.py

```diff
@@ -0,0 +1,42 @@
+import strax
+import numpy as np
+from straxen.plugins.peaks.peaks import Peaks
+
+export, __all__ = strax.exporter()
+
+
+@export
+class PeaksSOM(Peaks):
+    """Same as Peaks but include in addition SOM type field to be propagated to event_basics.
+
+    Thus, only change dtype.
+
+    """
+
+    __version__ = "0.0.1"
+    child_plugin = True
+
+    def infer_dtype(self):
+        peaklet_classification_dtype = self.deps["peaklet_classification"].dtype_for(
+            "peaklet_classification"
+        )
+        peaklets_dtype = self.deps["peaklets"].dtype_for("peaklets")
+        # The merged dtype is argument position dependent! It must be first classification then peaklet
+        # Otherwise strax will raise an error when checking for the returned dtype!
+        merged_s2s_dtype = strax.merged_dtype((peaklet_classification_dtype, peaklets_dtype))
+        return merged_s2s_dtype
+
+    def setup(self):
+        self.copy_function_name = "_copy_requested_peak_fields_som"
+
+    def compute(self, peaklets, merged_s2s):
+        result = super().compute(peaklets, merged_s2s)
+
+        # For merged S2s SOM and straxen type are undefined:
```

Comment:
Perhaps this is a really basic thing that I'm missing, but why are types undefined for merged S2s?

Reply:
Because SOM classifies only based on peaklets. Merged S2s is only called on S2s (`type==2`) and we assume that these merged peaks stay S2s (`type==2`). However, for each S2 there are multiple SOM sub-types and it is currently not defined which sub-type should be inherited to the merged S2s after merging. We could do something like the sub-type of the peak with the largest area, but not sure if this make sense. 
If one uses straxen types as classification for peaks then of course `type == straxen_type` and the `straxen_type` of MergedS2s would be defined again, however in that case the information is trivial to retrieve. So I preferred to not add an extra if case and make the code more complicated.